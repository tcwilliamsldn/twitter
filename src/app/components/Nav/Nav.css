.nav {
    height: 80px;
    border-bottom: 1px solid #dedede;
    display: flex;
    align-items: center;
    justify-content: center;
    padding-left: 12px;

    @media (width > 768px) {
        justify-content: start;
        padding-left: 48px;
    }
}

.home,
.stats {
    font-size: 18px;
    font-weight: bold;
    text-decoration: none;
    color: black;
    margin-right: 40px;

    &::before {
        margin-right: 12px;
        display: inline-block;
        width: 18px;
        height: 18px;
    }

    &:hover {
        text-decoration: underline;
    }
}

.active {
    color: #306aff;
}

/* When using svgs in the before pseudoelement I can't change the fill
color, so I've swapped the svg here. I'd potentially rethink this approach
doing the task again */
.home::before {
    content: url(./icons/home.svg);
}

.home.active::before {
    content: url(./icons/home-active.svg);
}

.stats::before {
    content: url(./icons/stats.svg);
}

.stats.active::before {
    content: url(./icons/stats-active.svg);
}
