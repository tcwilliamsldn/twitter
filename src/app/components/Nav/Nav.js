import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Nav.css';

const Nav = ({ location }) => {
    const { pathname } = location;

    // It was unclear what should be the 'home' link, as the tweets returned
    // by the API are the user's own tweets rather than a feed as in the
    // designs. For simplicity I switch to the tweet feed if the user is on
    // the stats page, or the stats page if the user is in the feed.

    const statsLink = pathname.includes('/stats')
        ? pathname
        : `${pathname.replace(/\/\s*$/, '')}/stats`;

    const homeLink = pathname.includes('/stats')
        ? pathname.replace('/stats', '')
        : pathname;

    return (
        <nav className={styles.nav}>
            <NavLink
                exact
                to={homeLink}
                className={styles.home}
                activeClassName={styles.active}
            >
                Home
            </NavLink>
            <NavLink
                exact
                to={statsLink}
                className={styles.stats}
                activeClassName={styles.active}
            >
                Stats
            </NavLink>
        </nav>
    );
};

export default Nav;
