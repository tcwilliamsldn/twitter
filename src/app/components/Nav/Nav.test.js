import Nav from './Nav';
import React from 'react';
import { shallow } from 'enzyme';

describe('Nav', function() {
    it('renders the nav', function() {
        const component = shallow(<Nav location={{ pathname: '/' }} />);
        expect(component).toMatchSnapshot();
    });

    it('renders the correct links in nav from stats page', function() {
        const component = shallow(
            <Nav location={{ pathname: '/test/stats' }} />
        );
        expect(component).toMatchSnapshot();
    });

    it('renders the correct links in nav from home page', function() {
        const component = shallow(<Nav location={{ pathname: '/test/' }} />);
        expect(component).toMatchSnapshot();
    });
});
