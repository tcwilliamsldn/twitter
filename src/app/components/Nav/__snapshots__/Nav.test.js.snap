// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders the correct links in nav from home page 1`] = `
<nav
  className="nav"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="home"
    exact={true}
    to="/test/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="stats"
    exact={true}
    to="/test/stats"
  >
    Stats
  </NavLink>
</nav>
`;

exports[`Nav renders the correct links in nav from stats page 1`] = `
<nav
  className="nav"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="home"
    exact={true}
    to="/test"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="stats"
    exact={true}
    to="/test/stats"
  >
    Stats
  </NavLink>
</nav>
`;

exports[`Nav renders the nav 1`] = `
<nav
  className="nav"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="home"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="stats"
    exact={true}
    to="/stats"
  >
    Stats
  </NavLink>
</nav>
`;
